opencog/atoms/pattern/PatternLink.cc

```diff
@@ -23,6 +23,8 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#include <boost/range/algorithm/find_if.hpp>
```

Comment:
I've finally concluded that boost consists of some of the most poorly written, poorly-designed C++ code in the known universe. It would probably be a good idea to try to minimize the amount of boost utilities that we use. Its painful, creaky, bug-prone.